source/commands/ci/_tests/runner.test.ts

```diff
@@ -25,6 +29,44 @@ it("uses the source and platform from config", async () => {
   expect(executor.platform).toEqual(platform)
 })
 
+it("does not use GitHub Checks by default", async () => {
+  await runRunner(defaultAppArgs as SharedCLI, { platform, source })
+
+  // Pull the executor out of the call to runDangerSubprocess
+  const executor: Executor = mockRunDangerSubprocess.mock.calls[0][2]
+  expect(executor.ciSource).toEqual(source)
+  expect(executor.platform).toEqual(platform)
+  expect(executor.options.disableGitHubChecksSupport).toEqual(true)
+})
+
+it("uses GitHub Checks if requested", async () => {
+  const customArgs = {
+    ...defaultAppArgs,
+    useGithubChecks: true,
+  } as SharedCLI
+  await runRunner(customArgs, { platform, source })
+
+  // Pull the executor out of the call to runDangerSubprocess
+  const executor: Executor = mockRunDangerSubprocess.mock.calls[0][2]
+  expect(executor.ciSource).toEqual(source)
+  expect(executor.platform).toEqual(platform)
+  expect(executor.options.disableGitHubChecksSupport).toEqual(false)
+})
+
+it("does not use GitHub Checks if requested not to", async () => {
+  const customArgs = {
+    ...defaultAppArgs,
+    useGithubChecks: false,
+  } as SharedCLI
+  await runRunner(customArgs, { platform, source })
+
+  // Pull the executor out of the call to runDangerSubprocess
+  const executor: Executor = mockRunDangerSubprocess.mock.calls[0][2]
+  expect(executor.ciSource).toEqual(source)
+  expect(executor.platform).toEqual(platform)
+  expect(executor.options.disableGitHubChecksSupport).toEqual(true)
+})
+
 // TODO: This occasionally fails!
```

Comment:
I suspect that this intermittency was caused by the failure to reset the mock but I didn't want to muddy up this PR with that change. If you concur with that assessment I'll raise a PR unskipping this after the mock resetting has landed. 🙂

Reply:
This makes sense to me - good thinking!